box64-bundle-x86-libs.sh

```diff
@@ -1,121 +1,123 @@
-#!/bin/bash
+#!/bin/sh
 
-set -x
+set -ex
 
-extract_pkg_auto() {
-    # Example x86_64 package name: libcurl3-gnutls_7.74.0-1.3+deb11u7~bpo11+1_amd64.deb
-    # Example x86 package name: libcurl3-gnutls_7.74.0-1.3+deb11u7~bpo11+1_i386.deb
-    if echo "${1}" | grep -q -P ".deb$"; then
-        echo "DEB (Debian) package detected."
-        ar x "$(basename "${1}")"
-        tar --extract --file data.tar*
-    # Example x86_64 package name: curl-gnutls-8.1.2-44-1-x86_64.eopkg
-    # Example x86 package name: curl-gnutls-32bit-8.1.2-44-1-x86_64.eopkg
-    elif echo "${1}" | grep -q -P ".eopkg$"; then
-        echo "EOPKG (Solus Linux) package detected."
-        unzip -o "$(basename "${1}")"
-        tar --extract --file install.tar.xz
-    # Example x86_64 package name: bluez-libs-5.63-1.el8.x86_64.rpm
-    # Example x86 package name: bluez-libs-5.63-1.el8.i686.rpm
-    elif echo "${1}" | grep -q -P ".rpm$"; then
-        echo "RPM (Fedora) detected."
-        rpm2cpio "$(basename "${1}")" | cpio -idmv
-    # Example x86_64 package name: libjpeg-turbo-3.0.1_1.x86_64.xbps
-    # Example x86 package name: libjpeg-turbo-32bit-3.0.1_1.x86_64.xbps
-    elif echo "${1}" | grep -q -P ".xbps$"; then
-        echo "XBPS (Void Linux) package detected."
-        tar --extract --file "$(basename "${1}")"
-    else
-        echo "Unsupported package type detected."
-        exit 1
-    fi
+trap cleanup_exit INT TERM EXIT
+
+cleanup_exit()
+{
+    echo "Running cleanup_exit..."
+    rm -rf /tmp/box64-bundle.*
 }
 
-box64_dir=$(pwd)
-dir_tmp_local="$(mktemp --directory)"
-cd "${dir_tmp_local}"
+# we must have few tools
+awk=$(which awk) || { echo "E: You must have awk" && exit 1; }
+curl=$(which curl) || { echo "E: You must have curl" && exit 1; }
+curl_cmd="${curl} --connect-timeout 5 --retry 5 --retry-delay 1 --create-dirs -fsSLo"
+rpm2cpio=$(which rpm2cpio) || { echo "E: You must have rpm2cpio" && exit 1; }
```

Comment:
As strange as it may sound, we also need a test for `cpio` itself. On the Alma Linux 9 minimal container image, `rpm2cpio` is installed but `cpio` is not. I assume that it is part of some embedded library for DNF and/or RPM which is why the CLI tool is not installed.

Reply:
ack, I'll

Reply:
done